src/secp256k1.c

```diff
@@ -502,6 +515,10 @@ static int secp256k1_ecdsa_sign_inner(const secp256k1_context* ctx, secp256k1_sc
     if (noncefp == NULL) {
         noncefp = secp256k1_nonce_function_default;
     }
+    /* sign-to-contract commitments only work with the default nonce function,
+     * because we need to ensure that s2c_data is actually hashed into the nonce and
+     * not just ignored. */
+    VERIFY_CHECK(s2c_data32 == NULL || noncefp == secp256k1_nonce_function_default);
```

Comment:
I believe I'm missing something obvious but don't understand this. Isn't the nonce derivation function (of the inner nonce) unrelated to the s2c commitment?

Reply:
No, because if the inner nonce is the same for two differing s2c commitments (in two signatures which have the same key/message), this will leak the secret nonce.

Reply:
Ok, I see. But maybe add to the comment that this is the reason that `s2c_data` should be hashed. This is different than the usual "sign two different messages with the same nonce" thing.